app/src/main/java/io/intrepid/contest/base/BaseMvpActivity.kt

```diff
@@ -0,0 +1,68 @@
+package io.intrepid.contest.base
+
+import android.content.Intent
+import android.os.Bundle
+import android.support.annotation.CallSuper
+
+/**
+ * Base class for activities that will have some business logic instead of just hosting a fragment.
+ * If the activity is only going to act as a container for a fragment, use [BaseFragmentActivity]
+ * instead
+ */
+abstract class BaseMvpActivity<P : BaseContract.Presenter<V>, V : BaseContract.View> : BaseActivity(), BaseContract.View {
```

Comment:
Interesting, were you forced to specify the view type, or did you add it for clarity/compat with Skotlinton?